plugins/pipelines/app/models/concerns/samson_pipelines/stage_concern.rb

```diff
@@ -1,7 +1,10 @@
 module SamsonPipelines::StageConcern
+  require 'byebug'
   # Return true if any stages in the pipeline are marked production
+  # make sure to check the no_code_deployed flag on the 
+  # next deployment as well
```

Comment:
original `production?` method does not check no_code_deployed ... so this is the wrong place to add this check